sigpy/mri/rf/slr.py

```diff
@@ -0,0 +1,648 @@
+        m = np.asarray([0, 0])
+        w = np.asarray([d1 / d2])
+        # first sub-band
+        ii = 1
+        Gcent = shift + (ii - G / 2 - 1 / 2) * tb / G  # first band center
+        # first band left edge
+        f = np.append(f, Gcent - (tb / G / 2 - ftw * (tb / 2)))
+        m = np.append(m, encode[ii - 1])
+        if encode[ii - 1] != encode[ii]:
+            # add the first band's right edge and its amplitude, and a weight
+            f = np.append(f, Gcent + (tb / G / 2 - ftw * (tb / 2)))
+            m = np.append(m, encode[ii - 1])
+            w = np.append(w, 1)
+        # middle sub-bands
+        for ii in range(2, G):
+            Gcent = shift + (ii - G / 2 - 1 / 2) * tb / G  # center of band
+            if encode[ii - 1] != encode[ii - 2]:
+                # add a left edge and amp for this band
+                f = np.append(f, Gcent - (tb / G / 2 - ftw * (tb / 2)))
+                m = np.append(m, encode[ii - 1])
+            if encode[ii - 1] != encode[ii]:
+                # add a right edge and its amp, and a weight for this band
+                f = np.append(f, Gcent + (tb / G / 2 - ftw * (tb / 2)))
+                m = np.append(m, encode[ii - 1])
+                w = np.append(w, 1)
+        # last sub-band
+        ii = G
+        Gcent = shift + (ii - G / 2 - 1 / 2) * tb / G  # center of last band
+        if encode[ii - 1] != encode[ii - 2]:
+            # add a left edge and amp for the last band
+            f = np.append(f, Gcent - (tb / G / 2 - ftw * (tb / 2)))
+            m = np.append(m, encode[ii - 1])
+        # add a right edge and its amp, and a weight for the last band
+        f = np.append(f, Gcent + (tb / G / 2 - ftw * (tb / 2)))
+        m = np.append(m, encode[ii - 1])
+        w = np.append(w, 1)
+        # right stop-band
+        f = np.append(f, (shift + (1 + ftw) * (tb / 2), (N / 2))) / (N / 2)
+        m = np.append(m, [0, 0])
+        w = np.append(w, d1 / d2)
+
+        # separate the positive and negative bands
+        mp = (m > 0).astype(float)
+        mn = (m < 0).astype(float)
+
+        # design the positive and negative filters
+        c = np.exp(1j * 2 * np.pi / (2 * (N + 1))
+                   * np.concatenate([np.arange(0, N / 2 + 1, 1),
+                                     np.arange(-N / 2, 0, 1)]))
+        bp = signal.firls(N + 1, f, mp, w)  # the positive filter
+        bn = signal.firls(N + 1, f, mn, w)  # the negative filter
+
+        # combine the filters and demodulate
+        b = np.fft.ifft(np.multiply(np.fft.fft(bp - bn), c))
+        b = np.real(b[:N])
+        # hilbert transform to suppress negative passband
+        b = signal.hilbert(b)
+        # demodulate to DC
+        cShift = np.exp(-1j * 2 * np.pi / N * shift * np.arange(0, N, 1)) / 2 \
+            * np.exp(-1j * np.pi / N * shift)
+        b = np.multiply(b, cShift)
+
+    return b
+
+
+def dzgSliderrf(N=256, G=5, flip=np.pi / 2, phi=np.pi, tb=12,
```

Comment:
Please change `dzgSliderrf` to `dzg_sliderrf`, and provide some documentation and reference.